pkg/chain/upkeep_observation.go

```diff
@@ -53,6 +56,9 @@ func (u *UpkeepObservation) validate(uo upkeepObservation) error {
 		if uiInt.String() != string(ui) {
 			return ErrInvalidUpkeepIdentifier
 		}
+		if uiInt.Cmp(big.NewInt(0)) == -1 {
```

Comment:
Should this be `<= 0` instead since zero value identifiers are also invalid?

Reply:
I considered it, technically upkeep ID is uint256 so it can be 0. I'll add it for the blockKey though

Reply:
Ahh. ok. Sounds good.